api/config/v1alpha1/metric_types.go

```diff
@@ -29,6 +37,19 @@ type MetricSink struct {
 	OpenTelemetry *OpenTelemetrySink `json:"openTelemetry,omitempty"`
 }
 
+type Match struct {
+	Type  MatcherType `json:"type"`
+	Value string      `json:"value"`
+}
+
```

Comment:
// MatcherType defines the stats matcher type
// 
// +kubebuilder:validation:Enum=RegularExpression;Prefix;Suffix